adagrams/game.py

```diff
@@ -1,11 +1,162 @@
+#Maggie T adagrams project 
+# Billie Betts adagram project
+
+import string
+import random
+
+quantity_data = {1: ['Z', 'X', 'K', 'Q', 'J'],
+                  2: ['B','C','F','H','M', 'P','V','W','Y'],
+                  3: ['G'],
+                  4: ['D', 'L', 'S', 'U'],
+                  6: ['N', 'R', 'T'],
+                  8: ['O'],
+                  9: ['A', 'I'],
+                  12: ['E']}
+
+value_data = {1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+              2: ['D','G'],
+              3: ['B', 'C', 'M', 'P'],
+              4: ['F', 'H', 'V', 'W', 'Y'],
+              5: ['K'],
+              8: ['J', 'X'],
+              10: ['Q', 'Z']}
+
+#Takes the two inputs and stores them in one dictionary 
+def generate_alphabet_data_list(quantity_data, value_data):
+  all_letters = string.ascii_uppercase
+  results = [] #dictionaries stored in results
+  for letter in range(len(all_letters)):
+    generate_dict = {}
+    generate_dict['letter'] = all_letters[letter]
+
+    for key, value in quantity_data.items():
+      if all_letters[letter] in value:
+        generate_dict['quantity'] = key
+
+    for key, value in value_data.items():
+      if all_letters[letter] in value:
+        generate_dict['value'] = key
+  
+    results.append(generate_dict)
+
+  return results
+  
+
+alphabet_data_list = generate_alphabet_data_list(quantity_data, value_data)
+#makes a list that generates the numbers of letters so its proportionate to the probability of getting a certain letter
+def letter_soup(letter_data):
+  letter_soup = []
+
+  for data in letter_data:
+    runner = 0
+    while runner < data['quantity']:
+      letter_soup.append(data['letter'])
+      runner += 1
+  
+  return letter_soup
+
+todays_soup = letter_soup(alphabet_data_list)
+#Returns letters to user
 def draw_letters():
-    pass
+  # letter_soup = []
+
+  # for data in alphabet_data_list:
+  #   runner = 0
+  #   while runner < data['quantity']:
+  #     letter_soup.append(data['letter'])
+  #     runner += 1
+
+  user_letter_pool = []
+  
+  while len(user_letter_pool) < 10:
+    draw = random.randint(0, (len(todays_soup)-1))
+    drawn_letter = todays_soup[draw]
+
+    if user_letter_pool.count(drawn_letter) < todays_soup.count(drawn_letter):
+      user_letter_pool.append(drawn_letter)
+
+  return user_letter_pool
 
+# Checks that letters being used in words are letters in a users letter bank
 def uses_available_letters(word, letter_bank):
-    pass
 
+    letter_count = {}
+  
+# loop through word, convert all letters to uppercase, add count to letter_count
+    for l in word:
+      letter = l.upper()
+      if letter not in letter_count:
+        letter_count[letter] = 0
+     
+      letter_count[letter] += 1
+      l_count = letter_count[letter] 
+
+      if l_count > letter_bank.count(letter):
+        return False
+  
+    return True
+
```

Comment:
You could simplify this function by making a copy of the. letter bank and iterating over the words. Once we are iterating over the words we can add a conditional to find out if the letter in the words is in the letter bank copy. Something like this:
```
for letter in word.upper():
        if letter in letter_bank_copy:
            letter_bank_copy.remove(letter)
        else:
            return False
```